addon/-private/sticky/table-sticky-polyfill.js

```diff
@@ -0,0 +1,105 @@
+/* global ResizeSensor */
+
+const TABLE_POLYFILL_MAP = new WeakMap();
+
+class TableStickyPolyfill {
+  constructor(element) {
+    this.element = element;
+
+    this.element.style.position = 'static';
+    this.side = element.tagName === 'THEAD' ? 'top' : 'bottom';
+
+    this.setupRaf = requestAnimationFrame(this.repositionStickyElements);
+
+    this.setupResizeSensors();
+    this.setupRowMutationObservers();
+
+    this.mutationObserver = new MutationObserver(() => {
```

Comment:
From what I understand, this observer reacts to every DOM event. In a complicated app, we would have lots of DOM events and this would be triggered multiple times. Why is using ResizeSensor not enough in this case?

Reply:
It reacts to specifically to children being added and removed, that's what the `childList: true` option does here. We need to reposition if a node was added or removed, because that node has not yet been positioned and could affect the positioning of other nodes.

This allows us to have a generic polyfill without tying into Ember directly.